gae_serve/README.md

```diff
@@ -0,0 +1,134 @@
+
+- [Google Cloud Platform SDK](https://cloud.google.com/sdk/).  The SDK includes the commandline tools `gcloud` for deploying the service and [`gsutil`](https://cloud.google.com/storage/docs/gsutil) for managing files on GCS.
+
+- A Google Cloud Platform project which as the following products enabled:
+
+    - [Google App Engine (GAE)](https://cloud.google.com/appengine/)
+
+    - [Google Cloud Storage (GCS)](https://cloud.google.com/storage/)
+
+    - [Cloud Endpoints (CE)](https://cloud.google.com/endpoints/)
+
+
+## Setup
+
+1. `git clone https://github.com/GoogleCloudPlatform/ml-on-gcp`
+
+1. `cd ml-on-gcp/gae_serve`
+
+1. The name `modelserve` (appearing in `app.yaml` and `modelserve.yaml`) is tentative which can be changed.  You should make the same change in the steps below.
+
+1. Update `modelserve.yaml`:  Replace `PROJECT_ID` with your GCP project's id in this line:
+
+    `host: "PROJECT_ID.appspot.com"`
+
+    * Note that this file defines the API specifying the input `X` to be an array of arrays of floats, and output `y` to be an array of floats.  The model included in this sample code `lr.pkl` is a pickled [linear regression model](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html) with 2-dimensional inputs.
+
+1. Deploy the service endpoint:
+
+    `gcloud service-management deploy modelserve.yaml`
+
+    This step deploys a [Cloud Endpoint](https://cloud.google.com/endpoints/) service, which allows us to monitor the API usage on the [Endpoints](https://console.cloud.google.com/endpoints) console page.
+
+1. If the deployment is successful, get the deployment's config id either from the [Endpoints](https://console.cloud.google.com/endpoints) console page under the service's Deployment history tab, or you can find all the configuration IDs by running the following:
+
+    `gcloud service-management configs list --service="PROJECT_ID.appspot.com"`
+
+    The configuration ID should look like `2017-08-03r0`.  The `r0, r1, ...` part in the configuration IDs indicate the revision numbers, and you should use the highest (most recent) revision number.
+
+1. Create a GCS bucket with your choice of a `BUCKET_NAME`, and copy the sample model file over:
+
+    ```
+    gsutil mb gs://BUCKET_NAME
+    gsutil cp lr.pkl gs://BUCKET_NAME
+    ```
+
+1. Update `app.yaml`:
+
+    * If you already have at least one GAE service in your GCP project:
+
+        - Replace `PROJECT_ID` with your GCP project's id.
+
+        - Replace `BUCKET_NAME` with the name of the bucket you created on GCS above.
+
+        - Replace `CONFIG_ID` with the configuration ID you got from the service endpoint deployment.
+
+    * If you have not deployed any service to GAE with this GCP project:
+
+        - Replace `service: modelserve` with `service: default` in `app.yaml`.  The first service deployed to GAE must be named `default`.
```

Comment:
*Temporarily* replace - can you specify a different YAML file when deploying to GAE? If so, would be good to include both YAML files in the repo.

Reply:
I hope to keep the structure of the sample in close to the same shape as would be for a setup in a real production system, where only two yaml files are needed (backend service and service endpoint), and so I hope to avoid adding an additional yaml and prefer forcing the user to edit the files and learn about the caveats.